### PR TITLE
New TLS policy 20200510 only TLS1.2

### PR DESCRIPTION
**Pull request by stephdl**
https://github.com/NethServer/dev/issues/6170

### CODE REVIEW

**Comment by stephdl** [comment]
@DavidePrincipi https://travis-ci.com/github/NethServer/nethserver-directory

```
spawn ssh-add /srv/makerpms/.ssh/id_rsa
Enter passphrase for /srv/makerpms/.ssh/id_rsa: 
Bad passphrase, try again for /srv/makerpms/.ssh/id_rsa: [UPLOAD_SHA1] nethserver-directory-3.3.3-1.2.pr50.g2923913.ns7.noarch 1 9d7cedc73fb257017aa506d49308d5f3e8290cfb
[UPLOAD_SHA1] nethserver-directory-3.3.3-1.2.pr50.g2923913.ns7.noarch (none) 59beb08790fb656a8210d7e3060b4d4197b69b13
Warning: Permanently added 'packages.nethserver.org,159.89.233.254' (ECDSA) to the list of known hosts.
Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
Couldn't read packet: Connection reset by peer
[ERROR] sftp upload error
```

File: root/etc/e-smith/events/actions/nethserver-directory-tlspolicy [hunk 1]
```diff
@@ -33,6 +33,17 @@ replace: olcTLSProtocolMin
...
+    ldapmodify -c -Y EXTERNAL &>/dev/null <<'EOF' || ((errors ++))
+dn: cn=config
+changetype: modify
+replace: olcTLSCipherSuite
+olcTLSCipherSuite: EECDH:EDH:CAMELLIA:ECDH:RSA:!eNULL:!SSLv2:!RC4:!DES:!EXP:!SEED:!IDEA:!3DES
+-
+replace: olcTLSProtocolMin
+olcTLSProtocolMin: 3.3
+EOF
+
```
Comment (DavidePrincipi) [inline root/etc/e-smith/events/actions/nethserver-directory-tlspolicy:46]: This should be the first IF branch, so you can simplify the IF checks:

```
if (( $tlspolicy >= 20200510 ))
    ...
elif (( $tlspolicy >= 20180621 ))
    ...
else
    ...
fi

Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-directory-tlspolicy]: done thank

File: root/etc/e-smith/events/actions/nethserver-directory-tlspolicy [hunk 2]
```diff
@@ -22,7 +22,18 @@
 
 tlspolicy=$(/sbin/e-smith/config getprop tls policy)
 
-if [[ -n ${tlspolicy} ]] && (( $tlspolicy >= 20180621 )); then
+if [[ -n ${tlspolicy} ]] && (( $tlspolicy >= 20200510 )); then
```
Comment (DavidePrincipi) [inline root/etc/e-smith/events/actions/nethserver-directory-tlspolicy:25]: Provide a default value if `tlspolicy` is an empty string:

```suggestion
if (( ${tlspolicy:-0} >= 20200510 )); then
```

File: root/etc/e-smith/events/actions/nethserver-directory-tlspolicy [hunk 3]
```diff
@@ -22,7 +22,18 @@
...
+dn: cn=config
+changetype: modify
+replace: olcTLSCipherSuite
+olcTLSCipherSuite: EECDH:EDH:CAMELLIA:ECDH:RSA:!eNULL:!SSLv2:!RC4:!DES:!EXP:!SEED:!IDEA:!3DES
+-
+replace: olcTLSProtocolMin
+olcTLSProtocolMin: 3.3
+EOF
+
+elif [[ -n ${tlspolicy} ]] && (( $tlspolicy >= 20180621 )); then
```
Comment (DavidePrincipi) [inline root/etc/e-smith/events/actions/nethserver-directory-tlspolicy:36]: Same here:

```suggestion
elif (( ${tlspolicy:-0} >= 20180621 )); then
```

File: root/etc/e-smith/events/actions/nethserver-directory-tlspolicy (hunk 2 above)
Comment (gsanchietti) [reply root/etc/e-smith/events/actions/nethserver-directory-tlspolicy:25]: I vote for Stephane implementation: is much more simple to read!

Comment (DavidePrincipi) [reply root/etc/e-smith/events/actions/nethserver-directory-tlspolicy:25]: I vote my review: it is much more shorter and beautiful :hugs:

Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-directory-tlspolicy:25]: Something new to learn I vote for your review

File: root/etc/e-smith/events/actions/nethserver-directory-tlspolicy (hunk 3 above)
Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-directory-tlspolicy]: done

File: root/etc/e-smith/events/actions/nethserver-directory-tlspolicy (hunk 2 above)
Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-directory-tlspolicy]: done

**Pull request merged**